src/sedpack/io/dataset_iteration.py

```diff
@@ -93,6 +100,19 @@ def shard_paths_dataset(
         if shards:
```

Comment:
what happens if `shards == 0`? It seems that you would take the full dataset?

Reply:
Yes. I do not see a huge benefit of having a dataset with no elements inside...
- We can explicitly document this.
- We can change the semantics to: `None` is ignored, integer is interpreted as `shard_list[:shards]` (same semantics would need to be with `custom_metadata_type_limit`).
WDYT?

Reply:
What I'm getting at is: passing `shards == 0` is very likely unintended, and I would error out. What I do in these situation is: 1) check for `if shards is not None` (rather than just `if shards`), 2) in case shards is not None, `if shards == 0: log.error("shards needs to be > 0") and sys.exit(1). Again, nitpick & personal preference, this is a very small thing.

Reply:
I agree this is not very intuitive. The question is however which is preferable:
1. Demand `shards > 0`.
2. Implement the semantics of Python lists with the possibility for omitting the last 10 shards by setting `shards=-10`. Possibly also noting the user that `shards==0` is probably a mistake.

Reply:
my 2cents: I don't see many reasonable use cases for (2), as such "take last 10" is meaningful only if "shards ordering" is meaningful (which I don't think it's the case in your scenario?). I would keep it simple, just request shards > 0; easier to implement & document.